cpp/src/parquet/column_reader.cc

```diff
@@ -263,6 +269,11 @@ class SerializedPageReader : public PageReader {
                                              int compressed_len, int uncompressed_len,
                                              int levels_byte_len = 0);
 
+  // Returns true for non-data pages, and if we should skip based on
+  // skip_page_callback_. Performs basic checks on values in the page header.
+  // Fills in page_statistics.
+  bool ShouldSkipPage(EncodedStatistics& page_statistics);
```

Comment:
nit: Style in Arrow is to still use pointers for output parameters

Reply:
Done